src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -713,6 +693,75 @@ public PathNavigate getNavigator()
         return newNavigator;
     }
 
+    @Override
+    public void entityInit()
+    {
+        super.entityInit();
+        dataManager.register(DATA_COLONY_ID, citizenColonyHandler == null ? 0 : citizenColonyHandler.getColonyId());
+        dataManager.register(DATA_CITIZEN_ID, citizenId);
+    }
+
+    /**
+     * Set the metadata for rendering.
+     *
+     * @param metadata the metadata required.
+     */
+    @Override
+    public void setRenderMetadata(final String metadata)
+    {
+        super.setRenderMetadata(metadata);
+        dataManager.set(DATA_RENDER_METADATA, getRenderMetadata());
+        //Display some debug info always available while testing
+        //tofo: remove this when in Beta!
+        //Will help track down some hard to find bugs (Pathfinding etc.)
+        if (citizenData != null)
+        {
+            if (citizenJobHandler.getColonyJob() != null && Configurations.gameplay.enableInDevelopmentFeatures)
+            {
+                setCustomNameTag(citizenData.getName() + " (" + citizenStatusHandler.getStatus() + ")[" + citizenJobHandler.getColonyJob().getNameTagDescription() + "]");
+            }
+            else
+            {
+                setCustomNameTag(citizenData.getName());
+            }
+        }
+    }
+
+    @Override
+    public void setCustomNameTag(@NotNull final String name)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)